test/src/e2e_vm_tests/test_programs/should_pass/language/many_stack_variables/src/main.sw

```diff
@@ -147,5 +147,186 @@ fn main() -> u64 {
     let b = ZERO_B256;
     let b = ZERO_B256;
 
-    return zzz.a + zzz.b + zzz.c + zzz.d
+    let k1 = 6;
+
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    foo();
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;
+    let c = 0;  
+    let c = 0; // problematic 
```

Comment:
This test still fails with this change. The failure used to be a memory overflow but now we just get the wrong result. There's something else going on that I'm still trying to figure out.

Reply:
Resolved.